src/LodeRunner/src/ValidationTest/Main.cs

```diff
@@ -390,8 +398,15 @@ public int RunLoop(Config config, CancellationToken token)
                 totalRequests += state.Count;
             }
 
-            // fire event
-            TestRunComplete(null, new LoadResultEventArgs(startTime, completedTime, config.TestRunId, (int)totalRequests, totalFailures));
+            if (token.IsCancellationRequested)
+            {
+                TestRunComplete(null, new LoadResultEventArgs(startTime, DateTime.UtcNow, config.TestRunId, 0, 0, SystemConstants.TestRunExecutionStoppedMessage));
```

Comment:
might be worth capturing requestcount and failure count inside cancellation event and leave it up to user/ui to determine how to use that data

Reply:
fixed